test/metabase/search/scenarios_test.clj

```diff
@@ -0,0 +1,54 @@
+(ns metabase.search.scenarios-test
+  (:require
+   [clojure.test :refer [deftest is testing]]
+   [metabase.config :as config]
+   [metabase.search.postgres.ingestion :as search.ingestion]
+   [metabase.search.spec :as search.spec]
+   [metabase.search.test-util :as search.tu]
+   [metabase.test :as mt]))
+
+(defn ingest!
+  "Ingest any objects with name contains `search-term`"
+  [search-term]
+  (doseq [model (keys (methods search.spec/spec))]
+    (#'search.ingestion/batch-update!
+     (#'search.ingestion/spec-index-reducible model [:like :this.name (str "%" search-term "%")]))))
+
+(defn fulltext-search
+  [search-string]
+  (map (juxt :model :id :name) (search.tu/search-results search-string {:search-engine "fulltext"})))
+
+(deftest e2e-accounts-test
```

Comment:
It would be nice if the test more closely represented the variation in the actual stats data, so that its less tied to our current rankers and their parameters.

For example:

- Using the actual view counts rather than numbers around our magic number 40.
- Having multiple text matches in some of the names, having descriptions, varying length of unique words in there.
- Having some pinned items (even though our weight was changed to 0)